client/components/payment-method-logos/index.tsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { Fragment } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	Visa,
+	MasterCard,
+	Amex,
+	WooPay,
+	ApplePay,
+	GooglePay,
+	Discover,
+	JCB,
+	Afterpay,
+	Affirm,
+	Klarna,
+} from './Icons';
+
+import './style.scss';
+
+const Payments = [
+	{
+		name: 'visa',
+		component: <Visa key="visa" />,
+	},
+	{
+		name: 'mastercard',
+		component: <MasterCard key="mastercard" />,
+	},
+	{
+		name: 'amex',
+		component: <Amex key="amex" />,
+	},
+	{
+		name: 'discover',
+		component: <Discover key="discover" />,
+	},
+	{
+		name: 'woopay',
+		component: <WooPay key="woopay" />,
+	},
+	{
+		name: 'applepay',
+		component: <ApplePay key="applepay" />,
+	},
+	{
+		name: 'googlepay',
+		component: <GooglePay key="googlepay" />,
+	},
+	{
+		name: 'afterpay',
+		component: <Afterpay key="afterpay" />,
+	},
+	{
+		name: 'affirm',
+		component: <Affirm key="affirm" />,
+	},
+	{
+		name: 'klarna',
+		component: <Klarna key="klarna" />,
+	},
+	{
+		name: 'jcb',
+		component: <JCB key="jcb" />,
+	},
+];
+
+export const WooPaymentMethodLogos: React.VFC< {
+	maxNrElements: number;
+} > = ( { maxNrElements = 10 } ) => {
+	let i = 0;
+	let j = 0;
+	return (
+		<>
+			<div className="woocommerce-payments-method-logos">
+				{ Payments.map( ( payment ) => {
+					if ( i >= maxNrElements ) {
+						return <Fragment key={ payment.name }></Fragment>;
+					}
+					i++;
+
+					return payment.component;
+				} ) }
+				{ i < 21 && (
+					<div className="woocommerce-payments-method-logos_count">
+						+ { 21 - i }
+					</div>
+				) }
+			</div>
+
+			<div className="woocommerce-payments-method-logos_mini">
```

Comment:
Same as above ^

+ Could you describe what we mean by `j` here? And why do we need a "logos_mini" version? It's not clear why we are displaying both 🤔 is the control (hide/show) done via CSS? Or?

Reply:
Just saw that the control is done via CSS, I'm not sure if this is the best way to do it given we'll have extra JS added to the page for no particular reason.

Reply:
it's just a "foreach". nothing that could affect the performance I think. do you think we could improve the performance with other approach ?

Reply:
It's not about the `foreach`, but the fact that we are adding extra svgs to the page, and then hiding them via CSS. This will make the page slower, so it's better to not add them in the first place.

Also, those SVGs are now adding +64.4 kB (+22%) to the JS file. So I think we should change the method here.

Reply:
I completely agree with @elazzabi. Instead of using CSS to hide different views, we should adjust the number of displayed logos based on the viewport width.